lib/puppet/provider/kernel_parameter/grub2.rb

```diff
@@ -175,14 +175,15 @@ def flush
       if FileTest.file?(c) || ( FileTest.symlink?(c) &&
           FileTest.directory?(File.dirname(File.absolute_path(File.readlink(c)))) )
 
-        cfg = c
-        break
+        cfg.push(c)
```

Comment:
More common syntax

```suggestion
        cfg << c
```

Reply:
Also, this can probably be refactored as `#select` instead of `#each` to avoid the push.

Reply:
I can't get the perl out of me.

Reply:
select is not immediately obvious due the super call.